youths/views.py

```diff
@@ -1,16 +1,48 @@
 from django.conf import settings
-from django.http import HttpResponse, JsonResponse
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers, status
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
 from youths.models import YouthProfile
 
 
-def profiles(request, *args, **kwargs):
-    if settings.GDPR_API_ENABLED:
-        # TODO: Add authentication and security
-        if request.method == "GET":
-            youth_profile = YouthProfile.objects.get(profile__pk=kwargs["id"])
-            return JsonResponse(youth_profile.serialize(), safe=False)
-        else:
-            return HttpResponse(status=405)
-    else:
-        return HttpResponse(status=404)
+class DryRunSerializer(serializers.Serializer):
+    dry_run = serializers.BooleanField(required=False, default=False)
+
+
+class YouthProfileGDPRAPIView(APIView):
+    # TODO: Add authentication and security
+
+    def dispatch(self, request, *args, **kwargs):
+        if not settings.GDPR_API_ENABLED:
+            return HttpResponse(status=404)
+        return super().dispatch(request, *args, **kwargs)
```

Comment:
If I'm reading this correctly, with this implementation it is not possible for a client to know whether a HTTP 404 response is caused by a non-existing youth profile, or the fact that the whole GDPR API is disabled, which I don't think is optimal.

FWIW the previous way of not including the view to urlpatterns at all when GDPR is disabled seemed like a simpler and more robust way of doing this, but I guess you had some good reason for the change 🙂

Reply:
I removed the setting from urlpatterns mainly because I had some issues with `pytest` and especially when running it with `--doctest-modules` (which is enabled in Travis). When run with that flag, the setting from `conftest` didn't have any effect on the urlpattern. Without that flag, the tests were passing just fine. :shrug: I thought it was less hackier than e.g. reloading the root urlconf for each test. 

I can still try to see if there's a nice way to make this work with having the setting on the urlpattern.

Reply:
Though just a side note `get_object_or_404` would have the following message `No YouthProfile matches the given query.`